src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -88,14 +88,30 @@ class BasicCrawlerOptions(TypedDict, Generic[TCrawlingContext]):
 
 
 class BasicCrawler(Generic[TCrawlingContext]):
-    """Provides a simple framework for parallel crawling of web pages.
-
-    The URLs to crawl are fed either from a static list of URLs or from a dynamic queue of URLs enabling recursive
-    crawling of websites.
-
-    `BasicCrawler` is a low-level tool that requires the user to implement the page download and data extraction
-    functionality themselves. If we want a crawler that already facilitates this functionality, we should consider using
-    one of its subclasses.
+    """A versatile web crawler for parallel URL fetching.
+
+    Provides a basic framework for crawling websites, allowing customization of request handling and data extraction.
+
+    Args:
+        request_provider (Optional[RequestProvider]): The provider for requests to be crawled.
+        request_handler (Optional[Function]): The function to handle individual requests.
+        http_client (Optional[BaseHttpClient]): The HTTP client for making web requests.
+        concurrency_settings (Optional[ConcurrencySettings]): Settings for controlling concurrency.
+        max_request_retries (int, optional): Maximum retry attempts for failed requests (default: 3).
+        max_requests_per_crawl (Optional[int]): Maximum number of requests per crawl (default: no limit).
+        max_session_rotations (int, optional): Maximum session rotations per request (default: 10).
+        configuration (Optional[Configuration]): Crawler configuration settings.
+        request_handler_timeout (timedelta, optional): Request handler timeout (default: 1 minute).
+        use_session_pool (bool, optional): Use session pool (default: True).
+        session_pool (Optional[SessionPool]): Preconfigured session pool.
+        retry_on_blocked (bool, optional): Retry on blocked requests (default: True).
+        proxy_configuration (Optional[ProxyConfiguration]): Proxy configuration.
+        statistics (Optional[Statistics]): Statistics object.
+        event_manager (Optional[EventManager]): Event manager.
+        configure_logging (bool, optional): Configure logging (default: True).
+        _context_pipeline (Optional[ContextPipeline[TCrawlingContext]]): Internal context pipeline.
+        _additional_context_managers (Optional[Sequence[AsyncContextManager]]): Additional context managers.
+        _logger (Optional[logging.Logger]): Logger instance.
```

Comment:
Please remove types from docstrings. They are redundant, as we have type annotations.